### ISSUE TITLE
llvm/clang 14 reports using clauses separately

### ISSUE THREAD

**Issue by pierrewillenbrockdfki** [CastXML/CastXML#220]
llvm/clang 14 started to use OutputUsingType to report on using clauses, where before it was reported the same way typedefs were(i assume). This manifests itself in some types from boost being missing in the xml output.

Here is a simple patch that uses OutputUsingType to generate the typedefs definitions castxml used to generate with llvm/clang 13.
[castxml-using.patch.txt](https://github.com/CastXML/CastXML/files/9962556/castxml-using.patch.txt)

**Comment by bradking** [comment]
Thanks for the investigation and patch!  Please post an example of a definition that's missed without this.

**Comment by bradking** [comment]
> before it was reported the same way typedefs were(i assume)

Please post an example where CastXML reports a typedef when built with Clang 13, but not with Clang 14+.

**Comment by pierrewillenbrockdfki** [comment]
Thank you for looking at this issue, i will get back to you with an example.

**Comment by pierrewillenbrockdfki** [comment]
First, i first saw this about 6 months ago, created my patch and then moved my work to an operating system that was not quite as cutting edge to carry llvm-15 (or -14, -13), so i may be misremembering some things. I seem to remember the original issue was with the `boost::uint32_t` definition, but i was unable to reproduce enough of the original setup to recreate the problem. Instead, i recreated the problem with __gnu_cxx::lldiv_t:

```
typedef struct
{
  long long int quot;
  long long int rem;
} lldiv_t;

namespace __gnu_cxx
{
  using ::lldiv_t;

  inline lldiv_t
  div(long long __n, long long __d)
  { lldiv_t __q; return __q; }

}
```

In the resulting xml file, when generating the gccxml compatible format with llvm 15, i find one instances similar to 

```
<Unimplemented id="_594" type_class="Using"/>
```

I see now, that with llvm 12, the `div` function returns the original Typedef, while with llvm 15, the "Unimplemented" type is returned, so one might consider my patch a workaround. I don't have an llvm 13 around for testing, i can try to have castxml linked against llvm 14 if you think that would be helpful. I seem to remember that i found that the `Using` class of AST types were added between llvm 13 and llvm 14.

**Comment by bradking** [comment]
Thanks.  I simplified the example to

```c++
struct A;
namespace start {
  using ::A;
  A f();
}
```

which one can run with `--castxml-output=1 --castxml-start start`.

When CastXML is built using Clang 13, the result is

```xml
<?xml version="1.0"?>
<CastXML format="1.3.0">
  <Namespace id="_1" name="start" context="_2" members="_3 _4"/>
  <Struct id="_3" name="A" context="_2" location="f1:1" file="f1" line="1" incomplete="1"/>
  <Function id="_4" name="f" returns="_3" context="_1" location="f1:4" file="f1" line="4" mangled="_ZN5start1fEv"/>
  <Namespace id="_2" name="::"/>
  <File id="f1" name="issue220.cpp"/>
</CastXML>
```

When CastXML is built using Clang 14, the result is

```xml
<?xml version="1.0"?>
<CastXML format="1.3.0">
  <Namespace id="_1" name="start" context="_2" members="_3 _4"/>
  <Struct id="_3" name="A" context="_2" location="f1:1" file="f1" line="1" incomplete="1"/>
  <Function id="_4" name="f" returns="_5" context="_1" location="f1:4" file="f1" line="4" mangled="_ZN5start1fEv"/>
  <Unimplemented id="_5" type_class="Using"/>
  <Namespace id="_2" name="::"/>
  <File id="f1" name="issue220.cpp"/>
</CastXML>
```

I'll look at restoring the behavior with Clang 14+.

**Comment by bradking** [comment]
Please try the fix in #221.  It also adds a minimal test case.
